sdk/python/feast/job_service.py

```diff
@@ -117,6 +126,22 @@ def StartStreamToOnlineIngestionJob(
         feature_table = self.client.get_feature_table(
             request.table_name, request.project
         )
+
+        if self.client._config.getboolean(CONFIG_JOB_SERVICE_ENABLE_CONTROL_LOOP):
```

Comment:
@tsotnet it seems like we can take two approaches with the whole job service structure.
1. A thin wrapper on the Feast Client. Don't access internal methods directly and maintain abstraction. Don't duplicate logic in two places (inside and outside the client). 
2. Provide a shared library as the backend for both the client and service. So the service doesnt have any client instances, but it creates its own config, launches its own jobs, etc. It still uses the same methods as the client, but it just doesnt call the client.

It seems like the current approach is a mix of these two. We pull in the client for config, and then we access internal methods directly. Are we concerned about the internal dependencies we are creating here?

Reply:
Not sure I agree with this. The launcher layer also extensively uses client, so they aren't clearly decoupled from each other. If we really want the good decoupling, not only should job service not use client, but launcher should also not use client at all. We can do that, but not sure this PR is the right place for it.

Reply:
> The launcher layer also extensively uses client, so they aren't clearly decoupled from each other.

Yes. I don't think it's a good idea to have the client in the launcher.

> If we really want the good decoupling, not only should job service not use client, but launcher should also not use client at all.

Well which approach do you think is preferrable? I am ok with the JS using the client, I just dont think we should mix the two approaches (use client and then bypass client).

> but not sure this PR is the right place for it.

Knowing that we don't want to increase the internal dependencies (bypassing Client), can we try and not make things worse with this PR at least?

Reply:
fyi for others: we discussed this in Zoom. Decided to move `ensure_stream_ingestion_jobs` in client and job service will call that. In a separate PR we will remove usage of client in launcher and replace it with config reader. In job service as well, we'll replace client with config object whenever possible.